Hi @bowlofeggs ,
I've added tests for my additions. Unfortunately, it caused one other test to fail.
`_____________ ERROR collecting bodhi/tests/client/test___init__.py _____________
bodhi/tests/client/test___init__.py:372: in <module>
    class TestQuery(unittest.TestCase):
bodhi/tests/client/test___init__.py:407: in TestQuery
    return_value=client_test_data.EXAMPLE_QUERY_MUNCH_MULTI, autospec=True)
E   AttributeError: 'module' object has no attribute 'EXAMPLE_QUERY_MUNCH_MULTI'`

I can't figure out why is it happening. Do you have any idea?

Thanks!

--- a/bodhi/client/__init__.py
+++ b/bodhi/client/__init__.py
@@ -427,6 +427,8 @@ def comment(update, text, karma, user, password, url, **kwargs):
 @updates.command()
 @click.option('--staging', help='Use the staging bodhi instance',
               is_flag=True, default=False)
+@click.option('--arch', help='Specify arch of packages to download, ' +
+              '"all" will retrieve packages from all architectures')
 @click.option('--cves', help='Download update(s) by CVE(s) (comma-separated list)')
 @click.option('--updateid', help='Download update(s) by ID(s) (comma-separated list)')
 @click.option('--builds', help='Download update(s) by build NVR(s) (comma-separated list)')
@@ -444,12 +446,17 @@ def download(url, **kwargs):
 
     Args:
         staging (bool): Whether to use the staging server or not.
+        arch (unicode): Requested architecture of packages to download.
+                        "all" will retrieve packages from all architectures.
         url (unicode): The URL of a Bodhi server to create the update on. Ignored if staging is
                        True.
         kwargs (dict): Other keyword arguments passed to us by click.
     """
     client = bindings.BodhiClient(base_url=url, staging=kwargs['staging'])
+    requested_arch = kwargs['arch']
+
     del(kwargs['staging'])
+    del(kwargs['arch'])
     # At this point we need to have reduced the kwargs dict to only our
     # query options (cves, updateid, builds)
     if not any(kwargs.values()):
@@ -476,8 +483,15 @@ def download(url, **kwargs):
                     # subprocess is icky, but koji module doesn't
                     # expose this in any usable way, and we don't want
                     # to rewrite it here.
-                    args = ('koji', 'download-build', '--arch=noarch',
-                            '--arch={0}'.format(platform.machine()), build['nvr'])
+                    if requested_arch is None:
+                        args = ('koji', 'download-build', '--arch=noarch',
+                                '--arch={0}'.format(platform.machine()), build['nvr'])
+                    else:
+                        if u'all' in requested_arch:
+                            args = ('koji', 'download-build', build['nvr'])
+                        if u'all' not in requested_arch:
+                            args = ('koji', 'download-build', '--arch=noarch',
+                                    '--arch={0}'.format(requested_arch), build['nvr'])
                     ret = subprocess.call(args)
                     if ret:
                         click.echo("WARNING: download of {0} failed!".format(build['nvr']))

--- a/bodhi/tests/client/test___init__.py
+++ b/bodhi/tests/client/test___init__.py
@@ -96,6 +96,49 @@ class TestDownload(unittest.TestCase):
 
     @mock.patch('bodhi.client.bindings.BodhiClient.csrf',
                 mock.MagicMock(return_value='a_csrf_token'))
+    @mock.patch('bodhi.client.bindings.BodhiClient.send_request',
+                return_value=client_test_data.EXAMPLE_QUERY_MUNCH, autospec=True)
+    @mock.patch('bodhi.client.subprocess.call', return_value=0)
+    def test_arch_flag(self, call, send_request):
+        """
+        Assert correct behavior with the --arch flag.
+        """
+        runner = testing.CliRunner()
+
+        result = runner.invoke(
+            client.download,
+            ['--builds', 'nodejs-grunt-wrap-0.3.0-2.fc25', '--arch', 'x86_64'])
+
+        self.assertEqual(result.exit_code, 0)
+        self.assertEqual(result.output,
+                         'Downloading packages from nodejs-grunt-wrap-0.3.0-2.fc25\n')
+        call.assert_called_once_with((
+            'koji', 'download-build', '--arch=noarch', '--arch=x86_64',
+            'nodejs-grunt-wrap-0.3.0-2.fc25'))
+
+    @mock.patch('bodhi.client.bindings.BodhiClient.csrf',
+                mock.MagicMock(return_value='a_csrf_token'))
+    @mock.patch('bodhi.client.bindings.BodhiClient.send_request',
+                return_value=client_test_data.EXAMPLE_QUERY_MUNCH, autospec=True)
+    @mock.patch('bodhi.client.subprocess.call', return_value=0)
+    def test_arch_all_flag(self, call, send_request):
+        """
+        Assert correct behavior with --arch all flag.
+        """
+        runner = testing.CliRunner()
+
+        result = runner.invoke(
+            client.download,
+            ['--builds', 'nodejs-grunt-wrap-0.3.0-2.fc25', '--arch', 'all'])
+
+        self.assertEqual(result.exit_code, 0)
+        self.assertEqual(result.output,
+                         'Downloading packages from nodejs-grunt-wrap-0.3.0-2.fc25\n')
+        call.assert_called_once_with((
+            'koji', 'download-build', 'nodejs-grunt-wrap-0.3.0-2.fc25'))
+
+    @mock.patch('bodhi.client.bindings.BodhiClient.csrf',
+                mock.MagicMock(return_value='a_csrf_token'))
     @mock.patch('bodhi.client.bindings.BodhiClient.send_request')
     def test_empty_options(self, send_request):
         """

--- a/docs/man_pages/bodhi.rst
+++ b/docs/man_pages/bodhi.rst
@@ -156,9 +156,14 @@ The ``updates`` command allows users to interact with bodhi updates.
 
         A comman-separated list of update IDs you would like to download.
 
-    ``--builds <nvrs``
+    ``--builds <nvrs>``
 
         A comma-separated list of NVRs that identify updates you would like to download.
+
+    ``--arch <arch>``
+
+        You can specify an architecture of packages to download. "all" will download packages for all architectures.
+        Omitting this option will download packages for the architecture you are currently running.
 
 ``bodhi updates new [options] <builds>``
 

